vaadin-grid-flow-parent/vaadin-grid-flow/src/main/java/com/vaadin/flow/component/grid/Grid.java

```diff
@@ -527,6 +527,50 @@ public Renderer<T> getRenderer() {
             return renderer;
         }
 
+        /**
+         * Switch the renderer used for this column.
+         *
+         * @param renderer
+         *            the new renderer to be used for this column, should never
+         *            be {@code null}
+         *
+         * @since 24.1
+         */
+        public void setRenderer(Renderer<T> renderer) {
+            this.renderer = Objects.requireNonNull(renderer);
+
+            destroyDataGenerators();
+
+            rendering = renderer.render(getElement(), (KeyMapper<T>) getGrid()
```

Comment:
Renderings are not properly cleaned up at the moment. There is a `Rendering.getRegistration()`, which should be called to dispose the old renderer. See `VirtualList.setRenderer` for an example.

---

Semi-related, `ComponentRenderer` is not properly cleaned up when removing its registration, the container element that it adds is not removed from the Flow `Column`s element.

Changing the integration test to:
```
        NativeButton swapRenderersButton = new NativeButton(
                "Swap price column renderer",
                clickEvent -> {
                    priceColumn.setRenderer(new ComponentRenderer<>(Span::new, (span, item) -> {
                        span.setText("$" + item.getPrice());
                    }));
                    int virtualChildrenSize = priceColumn.getElement().getNode().getFeature(VirtualChildrenList.class).size();
                    System.out.println("Column virtual children size: " + virtualChildrenSize);
                });
```

Logs on sub-sequent clicks:
```
Virtual children size: 2
Virtual children size: 3
Virtual children size: 4
Virtual children size: 5
Virtual children size: 6
Virtual children size: 7
```

While only indirectly related to this PR, I feel that should be resolved somehow, though ideally in a separate PR.